workspaces/spectacle/src/index.ts

```diff
@@ -216,6 +216,11 @@ export async function makeSpectacle(opticContext: IOpticContext) {
     return projections;
   }
 
+  opticContext.specRepository.notifications.on('change', () => {
+    console.count('reloading because of specRepository change');
+    reload(opticContext);
```

Comment:
This is a short coming of using EventEmitters for this, rather than a Stream: they don't allow for async handling. That, in turn, introduces a race here: multiple changes can cause multiple reloads in-flight, and a race over which one finishes. It's small, but once it's hit, it could make a real head scratcher of a bug (especially as mutations _are_ reading in the current spec).